cpuid/src/brand_string.rs

```diff
@@ -395,5 +440,17 @@ mod tests {
 
         let short_bstr = BrandString::from_bytes_unchecked(b"GHz");
         assert!(short_bstr.find_freq().is_none());
+
+        let multiple_points_bstr = BrandString::from_bytes_unchecked(b"50.5.20GHz");
+        assert_eq!(multiple_points_bstr.find_freq().unwrap(), b"5.20GHz");
+
+        let no_decimal_bstr = BrandString::from_bytes_unchecked(b"5GHz");
+        assert!(no_decimal_bstr.find_freq().is_none());
+
+        let interrupted_bstr = BrandString::from_bytes_unchecked(b"500.00M5.20GHz");
+        assert_eq!(interrupted_bstr.find_freq().unwrap(), b"5.20GHz");
+
+        let split_bstr = BrandString::from_bytes_unchecked(b"5.30AMHz");
```

Comment:
Add another string which should be correct, but ridiculous. Something like `1.12bc5.30MaHz2.4.25THz` which should be matched to `4.25Thz`.